### DISCUSSION TITLE
How do I set the default value of a checkbox with Ionic?

### DISCUSSION THREAD

**Discussion by ptmkenny** [react-hook-form/react-hook-form#4208] [Q&A] [answered]
In my React app, I have been using react-hook-form like this:

```
        <input
          type="checkbox"
          id="edit-field-bool-card-subscription-value"
          name="field_bool_card_subscription[value]"
          defaultChecked={checked}
          ref={register}
          className="form-checkbox is-link checkbox"
        />
```

This works great.

Now, though, I am trying to make an Ionic 5 app based on the same code.

Here's an Ionic checkbox component:

```
      <IonCheckbox
        id="edit-field-bool-email-reminders-value"
        color="primary"
        name="field_bool_email_reminders[value]"
        checked={checked}
        ref={register}
      />
```

The problem is that the checkbox isn't checked.  When I remove the `ref={register}`, then the box is checked as I expect.  So I need another way to tell react-hook-form to check the box.

Based on the documentation, I tried setting the default values for the form in `useForm()`:

```
  const { register, handleSubmit, errors } = useForm({
    defaultValues: {
      email: userObject.emailAddress,
      field_bool_card_subscription: userObject.cardSubscriptionBool,
```

Unfortunately, this works for my text inputs (like email) but not for my checkboxes (the card subscription).

How can I set the default value of a checkbox in the `useForm` default values?  Or is there another way to set the default value of a checkbox when using an Ionic checkbox component?

**Comment by bluebill1049** [comment]
please use useController/Controller with ionic components, as the `ref` exposed from iconic is not the actual `input` DOM

there is a useful resource here for iconic components: https://react-hook-form.com/resources

**Comment by ptmkenny** [comment, marked as answer]
Thanks!  Based on your help, this is how I got it working:

I needed to add `[value]` in the defaultValues for the checkbox:

```
  const { register, handleSubmit, errors } = useForm({
    defaultValues: {
      email: userObject.emailAddress,
      'field_bool_card_subscription[value]': userObject.cardSubscriptionBool,
```

And then wrap the checkbox in the controller:

```
      <Controller
        render={({ onChange, value }) => (
          <IonCheckbox
            id="edit-field-bool-card-subscription-value"
            slot="start"
            checked={value}
            onIonChange={(e: CustomEvent<CheckboxChangeEventDetail>) => {
              onChange(e?.detail.checked);
            }}
          />
        )}
        control={control}
        name="field_bool_card_subscription[value]"
      />
```

And the interface:

```
export interface CheckboxChangeEventDetail {
  value: boolean;
  checked: boolean;
}
```

More details here: https://stackoverflow.com/a/63082568

**Reply by bluebill1049** [reply]
nice.
